### PR TITLE
add the telemetry to the default list of dataplane-operator services

### PR DESCRIPTION
**Pull request by jlarriba**

### CODE REVIEW

**Comment by jlarriba** [comment]
This depends on PR https://github.com/openstack-k8s-operators/edpm-ansible/pull/312 to merge.

**Comment by slagle** [comment]
/hold

We are holding dataplane-operator PRs until the CI stabilizes after merging https://github.com/openstack-k8s-operators/dataplane-operator/pull/303. The /hold will be dropped here once all PRs are in to support 303.

**Comment by slagle** [comment]
/unhold

rdoproject.org/github-check jobs should be passing now.

**Review by slagle — COMMENTED** [review]
+2

**Review by fao89 — APPROVED** [review]

**Comment by fao89** [comment]
recheck

**Comment by fao89** [comment]
Actually, it seems a legit error:
https://logserver.rdoproject.org/92/392/d745e1167b79e620f901fdeca7d75dc655d66c05/github-check/podified-multinode-edpm-deployment-crc/566c5a4/controller/ci-framework-data/logs/crc/pods/openstack_dataplane-deployment-telemetry-openstack-edpm-bnnv5_ad8ed579-5173-475e-b0e9-cd68d65184b9/openstackansibleee/0.log

**Comment by slagle** [comment]
The CI failure here needs to be fixed, I think it's on the edpm-ansible side.

From:
https://logserver.rdoproject.org/92/392/d745e1167b79e620f901fdeca7d75dc655d66c05/github-check/podified-multinode-edpm-deployment-crc/5e2e43d/controller/ci-framework-data/artifacts/must-gather.local.8574441275466136731/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-070a98749455b097c66a1a25ff827a3c0ee7385f1f0c8bf54f5b15438501f2bd/namespaces/openstack/pods/dataplane-deployment-telemetry-openstack-edpm-kfgz7/logs/dataplane-deployment-telemetry-openstack-edpm-kfgz7.log

```
[pod/dataplane-deployment-telemetry-openstack-edpm-kfgz7/openstackansibleee] [0;31mfailed: [edpm-compute-0] (item={'src': '/var/lib/config-data/merged/ceilometer.conf', 'dest': '/var/lib/openstack/config/ceilometer/ceilometer.conf'}) => {"ansible_loop_var": "item", "changed": false, "item": {"dest": "/var/lib/openstack/config/ceilometer/ceilometer.conf", "src": "/var/lib/config-data/merged/ceilometer.conf"}, "msg": "Could not find or access '/var/lib/config-data/merged/ceilometer.conf' on the Ansible Controller.\nIf you are using a module and expect the file to exist on the remote, see the remote_src option"}[0m
```

**Comment by jlarriba** [comment]
Yeah, ansibleEE is not mounting the configMaps with the configuration of the ceilometer compute and the infra compute. I have to introduce a modification in the telemetry-operator to make the ceilometer central create the configmaps. I will link the PR here.

**Comment by jlarriba** [comment]
This PR introduces the necessary change in telemetry-operator: https://github.com/openstack-k8s-operators/telemetry-operator/pull/179. It entails an API name change, I am working my way to get it integrated with openstack-operator: https://github.com/openstack-k8s-operators/openstack-operator/pull/466

**Comment by jlarriba** [comment]
This is almost done. I have switched the telemetry-operator to use Secrets instead of ConfigMaps in https://github.com/openstack-k8s-operators/telemetry-operator/pull/189. I have submitted a PR to openstack-operator to update it to use the latest telemetry version in https://github.com/openstack-k8s-operators/openstack-operator/pull/476 (which is failing the tests for something not related with my change). A change in edpm-ansible was also needed and it is almost merged: https://github.com/openstack-k8s-operators/edpm-ansible/pull/350.

The current error is cause by using an old version of telemetry-operator in https://logserver.rdoproject.org/92/392/15be7063e278641a5cb5f5cef0bac89cf592ce63/github-check/podified-multinode-edpm-deployment-crc/27c3785/controller/ci-framework-data/artifacts/must-gather.local.2327200559420966705/quay-io-openstack-k8s-operators-openstack-must-gather-sha256-070a98749455b097c66a1a25ff827a3c0ee7385f1f0c8bf54f5b15438501f2bd/namespaces/openstack-operators/pods/dataplane-operator-controller-manager-5c58f98fdb-kdnqc/logs/dataplane-operator-controller-manager-5c58f98fdb-kdnqc.log

`[pod/dataplane-operator-controller-manager-5c58f98fdb-kdnqc/manager] {"level":"error","ts":"2023-09-19T16:14:31Z","msg":"OpenStackDeployment error for NodeSet openstack-edpm","controller":"openstackdataplanedeployment","controllerGroup":"dataplane.openstack.org","controllerKind":"OpenStackDataPlaneDeployment","OpenStackDataPlaneDeployment":{"name":"openstack-edpm","namespace":"openstack"},"namespace":"openstack","name":"openstack-edpm","reconcileID":"8d299de2-9286-4793-8589-6a111b7bcb41","ObjectType":"*v1beta1.OpenStackDataPlaneDeployment","ObjectNamespace":"openstack","ObjectName":"openstack-edpm","error":"Secret \"ceilometer-compute-config-data\" not found","stacktrace":"github.com/openstack-k8s-operators/lib-common/modules/common/util.LogErrorForObject\n\t/go/pkg/mod/github.com/openstack-k8s-operators/lib-common/modules/common@v0.1.1-0.20230913075424-2680ce4b6ad2/util/log.go:65\ngithub.com/openstack-k8s-operators/dataplane-operator/controllers.(*OpenStackDataPlaneDeploymentReconciler).`

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_with_ipam.yaml
Comment (slagle) [inline config/samples/dataplane_v1beta1_openstackdataplanenodeset_with_ipam.yaml:55]: Oops!

**Review by slagle — CHANGES_REQUESTED** [review]

File: config/samples/dataplane_v1beta1_openstackdataplanenodeset_with_ipam.yaml
Comment (jlarriba) [reply config/samples/dataplane_v1beta1_openstackdataplanenodeset_with_ipam.yaml]: ooops :(

**Comment by jlarriba** [comment]
recheck

**Review by fao89 — APPROVED** [review]

**Comment by jlarriba** [comment]
The error is:

Could not find or access '/var/lib/openstack/config/telemetry/polling.yaml' on the Ansible Controller

While the polling.yaml file is mounted in /var/lib/openstack/configs/telemetry/polling.yaml. Small typo, look at the config vs configs.

I have submitted the appropiate fix to edpm_ansible: https://github.com/openstack-k8s-operators/edpm-ansible/pull/368/files

**Comment by rabi** [comment]
recheck

Looks like  https://github.com/openstack-k8s-operators/edpm-ansible/pull/368 has been merged.

**Comment by rabi** [comment]
This still looks broken https://logserver.rdoproject.org/92/392/a28cb5b2d44b969e52d5102d0e5b4efaa3d60323/github-check/cifmw-crc-podified-edpm-baremetal/02d24fb/ci-framework-data/logs/crc/pods/openstack_dataplane-deployment-telemetry-openstack-edpm-ipam-f4xbq_2435efdd-17df-4155-86d5-9963281ce717/openstackansibleee/0.log

**Comment by jlarriba** [comment]
Indeed, working on it

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
Working on the error

**Comment by jlarriba** [comment]
podman_containers is presenting a strange behaviour when executed in ansibleEE containers. This fails because node_exporter is not being created. According to the logs:

```
Sep 26 06:13:59 np0004183008 podman[111824]: 2023-09-26 10:13:59.871219738 +0000 UTC m=+0.052653626 container remove 75d88fed233c08224028184bc4872aff498e0afbfdf7f7627c226831875b8bd5 (image=quay.io/prometheus/node-exporter:v1.5.0, name=node_exporter, maintainer=The Prometheus Authors <prometheus-developers@googlegroups.com>)
Sep 26 06:13:59 np0004183008 python3[111766]: ansible-containers.podman.podman_container PODMAN-CONTAINER-DEBUG: podman rm -f node_exporter
Sep 26 06:13:59 np0004183008 systemd[3966]: Started podman-111837.scope.
Sep 26 06:13:59 np0004183008 podman[111837]: 2023-09-26 10:13:59.94917157 +0000 UTC m=+0.058407299 container create 375c510283d02c7ccb6e088c5d6563df3d06e083a576b6790e248df00c83d331 (image=quay.io/prometheus/node-exporter:v1.5.0, name=node_exporter, maintainer=The Prometheus Authors <prometheus-developers@googlegroups.com>)
Sep 26 06:13:59 np0004183008 podman[111837]: 2023-09-26 10:13:59.926524629 +0000 UTC m=+0.035760348 image pull 0da6a335fe1356545476b749c68f022c897de3a2139e8f0054f6937349ee2b83 quay.io/prometheus/node-exporter:v1.5.0
Sep 26 06:13:59 np0004183008 python3[111766]: ansible-containers.podman.podman_container PODMAN-CONTAINER-DEBUG: podman create --name node_exporter --publish 9100:9100 quay.io/prometheus/node-exporter:v1.5.0
```

As we can see, it is running a podman rm -f node_exporter and then a creation. When I run the same playbook using standalone ansible, the podman rm does not happen and the container is happily created.

As we are in the verge of DP2, I have submitted a PR (https://github.com/openstack-k8s-operators/edpm-ansible/pull/372) to edpm_ansible to replace the usage of podman_containers and use command instead until we can figure out what configuration in ansibleEE is making this behave differently.

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
This is still broken. Now, with the new code to just create node_exporter, the log says:

```
[pod/dataplane-deployment-telemetry-openstack-edpm-qdpp2/openstackansibleee] TASK [osp.edpm.edpm_telemetry : deploy node_exporter container] ****************
[pod/dataplane-deployment-telemetry-openstack-edpm-qdpp2/openstackansibleee] Wednesday 27 September 2023  11:30:27 +0000 (0:00:01.376)       0:00:26.382 *** 
[pod/dataplane-deployment-telemetry-openstack-edpm-qdpp2/openstackansibleee] [0;31mfatal: [edpm-compute-0]: FAILED! => {"changed": true, "cmd": ["podman", "create", "--name", "node_exporter", "--publish", "9100:9100", "quay.io/prometheus/node-exporter:v1.5.0"], "delta": "0:00:00.090099", "end": "2023-09-27 11:30:28.070138", "msg": "non-zero return code", "rc": 125, "start": "2023-09-27 11:30:27.980039", "stderr": "Error: creating container storage: the container name \"node_exporter\" is already in use by 99cf83813c820394b589c87530fa30aa577c9fd2b11849021a859abb4a62fd49. You have to remove that container to be able to reuse that name: that name is already in use", "stderr_lines": ["Error: creating container storage: the container name \"node_exporter\" is already in use by 99cf83813c820394b589c87530fa30aa577c9fd2b11849021a859abb4a62fd49. You have to remove that container to be able to reuse that name: that name is already in use"], "stdout": "", "stdout_lines": []}[0m
```

How is it possible that node_exporter container is already running if we have not deployed it before? Is the test also testing ansible idempotence?

However, I created a new PR on edpm_ansible to achieve the playbook idempotence: https://github.com/openstack-k8s-operators/edpm-ansible/pull/377

**Comment by jlarriba** [comment]
recheck

**Comment by jlarriba** [comment]
recheck

**Pull request merged**
